lib/api/apiUtils/bucket/models/ReplicationConfiguration.js

```diff
@@ -114,24 +114,40 @@ class ReplicationConfiguration {
     }
 
     /**
+     * Check if the Role field of the replication configuration is valid
+     * @param {string} ARN - The Role field value provided in the configuration
+     * @return {boolean} `true` if a valid role ARN, `false` otherwise
+     */
+    _isValidRoleARN(ARN) {
+        // AWS accepts a range of values for the Role field. Though this does
+        // not encompass all constraints imposed by AWS, we have opted to
+        // enforce the following.
+        const arr = ARN.split(':');
+        const isValidRoleARN =
+            arr[0] === 'arn' &&
+            arr[1] === 'aws' &&
+            arr[2] === 'iam' &&
+            arr[3] === '' &&
+            (arr[4] === '*' || arr[4].length > 1) &&
+            arr[5].startsWith('role');
+        return isValidRoleARN;
+    }
+
+    /**
      * Check that the `Role` property of the configuration is valid
      * @return {undefined}
      */
     _parseRole() {
-        const Role = this._parsedXML.ReplicationConfiguration.Role;
+        const Role = this._parsedXML.ReplicationConfiguration.Role &&
```

Comment:
Agree. I've split this into two variables for clarity and added a test for a Role value of `''`. In that case it should give an `InvalidArgument` error, not `MalformedXML`.